### PR TITLE
feat: schedule polling event emitter (CMC-793)

### PR DESCRIPTION
**Pull request by pd-dudkiewicz**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-793

### Change description ###

- scheduler for polling event emitter, disabled on preview
- added missing bpmn tests.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/unspec/bpmn/AcknowledgeServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+class AcknowledgeServiceTest extends BpmnBaseTest {
+
+    public static final String TOPIC_NAME = "processCaseEvent";
+
+    public AcknowledgeServiceTest() {
+        super("camunda/acknowledge_service.bpmn", "ACKNOWLEDGE_SERVICE");
```
Comment (swalker125) [inline src/test/java/uk/gov/hmcts/reform/unspec/bpmn/AcknowledgeServiceTest.java]: can we add /camunda in base class?

File: src/test/java/uk/gov/hmcts/reform/unspec/bpmn/BpmnBaseTest.java [hunk 2]
```diff
@@ -83,6 +84,19 @@ static void shutDown() {
         return engine.getExternalTaskService().createExternalTaskQuery().list();
     }
 
+    /**
+     * Retrieves a process definition which has a start message event with the messageName.
+     *
+     * @param messageName the name of the message.
+     * @return process definitions with given message start message event.
+     */
+    public ProcessDefinition getProcessDefinitionByMessage(String messageName) {
```
Comment (swalker125) [inline src/test/java/uk/gov/hmcts/reform/unspec/bpmn/BpmnBaseTest.java]: worth enumerating messageName?

File: src/test/java/uk/gov/hmcts/reform/unspec/bpmn/CaseStayedSchedulerTest.java [hunk 3]
```diff
@@ -38,10 +38,7 @@ void caseStayedSchedulerBmpnShouldFireCaseStayedExternalTask_whenStarted() {
 
         //get external tasks
         List<ExternalTask> externalTasks = getExternalTasks();
-
-        //assert task is as expected
         assertThat(externalTasks).hasSize(1);
-        assertThat(externalTasks.get(0).getTopicName()).isEqualTo("CASE_STAYED_FINDER");
```
Comment (swalker125) [inline src/test/java/uk/gov/hmcts/reform/unspec/bpmn/CaseStayedSchedulerTest.java]: intentional delete?

File: src/test/java/uk/gov/hmcts/reform/unspec/bpmn/ClaimantResponseTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    @Test
+    void caseEventTaskShouldFireCaseEventExternalTask_whenStarted() {
+        //assert process has started
+        assertFalse(processInstance.isEnded());
+
+        //assert topic names
+        assertThat(getTopics()).containsOnly(TOPIC_NAME);
+
+        //assert message start event
+        assertThat(getProcessDefinitionByMessage("CLAIMANT_RESPONSE")).isNotNull();
```
Comment (swalker125) [inline src/test/java/uk/gov/hmcts/reform/unspec/bpmn/ClaimantResponseTest.java]: what does isNotNull tell us here? Can we assert that correct actvityId is started or something?

Comment (swalker125) [reply src/test/java/uk/gov/hmcts/reform/unspec/bpmn/ClaimantResponseTest.java]: Few other places this is done

File: src/test/java/uk/gov/hmcts/reform/unspec/bpmn/AcknowledgeServiceTest.java (hunk 1 above)
Comment (pd-dudkiewicz) [reply src/test/java/uk/gov/hmcts/reform/unspec/bpmn/AcknowledgeServiceTest.java]: done

File: src/test/java/uk/gov/hmcts/reform/unspec/bpmn/BpmnBaseTest.java (hunk 2 above)
Comment (pd-dudkiewicz) [reply src/test/java/uk/gov/hmcts/reform/unspec/bpmn/BpmnBaseTest.java]: We could use CaseEvents here, which most likely won't change, but I'm not sure if we want to have that dependency. I'd leave it as string, the worst thing that can happen is that definition won't be found.

File: src/test/java/uk/gov/hmcts/reform/unspec/bpmn/CaseStayedSchedulerTest.java (hunk 3 above)
Comment (pd-dudkiewicz) [reply src/test/java/uk/gov/hmcts/reform/unspec/bpmn/CaseStayedSchedulerTest.java]: yes, we already have assertions for topic name:
`assertThat(getTopics()).containsOnly(TOPIC_NAME);`
so this line felt redundant

File: src/test/java/uk/gov/hmcts/reform/unspec/bpmn/ClaimantResponseTest.java (hunk 4 above)
Comment (pd-dudkiewicz) [reply src/test/java/uk/gov/hmcts/reform/unspec/bpmn/ClaimantResponseTest.java]: It was telling us that there is a process definition triggered by given message. But added checking if process id is correct.

**Review by dharmendrak — DISMISSED** [review]

**Review by swalker125 — COMMENTED** [review]
Just before I review, is there a reason we haven't gone down the path of one email handler, when camunda supplies the email template reference and we build the data from the reference?

**Comment by pd-dudkiewicz** [comment]
> Just before I review, is there a reason we haven't gone down the path of one email handler, when camunda supplies the email template reference and we build the data from the reference?

@swalker125 I think the idea was that we have one generic external task handler for all ccd events triggered by camunda, where we pass only ccd id and case event id. Also, sending template reference itself wouldn't help too much, because we want to trigger different ccd events for different notifications (in case we'd like to show that it was triggered on the UI).

**Review by douglasrice — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

**Pull request merged**